controllers/storageclient_controller.go

```diff
@@ -348,8 +348,8 @@ func (s *StorageClientReconciler) verifyNoStorageClassClaimsExist(instance *v1al
 
 	for i := range storageClassClaims.Items {
 		storageClassClaim := &storageClassClaims.Items[i]
-		sc := storageClassClaim.Labels[storageClientLabel]
-		if sc == instance.Name {
+		sc := storageClassClaim.GetAnnotations()
+		if sc[storageClientAnnotationKey] == client.ObjectKeyFromObject(instance).String() {
```

Comment:
Can't we instead add an indexer for the annotation and list by that indexer?

Reply:
done.